aea/configurations/base.py

```diff
@@ -225,7 +225,11 @@ class PackageConfiguration(Configuration, ABC):
 
     default_configuration_filename: str
     package_type: PackageType
+
     FIELDS_ALLOWED_TO_UPDATE: FrozenSet[str] = frozenset(["build_directory"])
+    FIELDS_WITH_NESTED_FIELDS: FrozenSet[str] = frozenset()
+    NESTED_FIELDS_ALLOWED_TO_UPDATE: FrozenSet[str] = frozenset()
+
```

Comment:
Something, that I don't like about this approach after looking at the code, is `NESTED_FIELDS_ALLOWED_TO_UPDATE` because this variable does not make sense by the definition of it. I don't think we should restrict the updates on nested fields because this will make things hard moving forward

Reply:
https://github.com/valory-xyz/open-aea/blob/main/aea/configurations/base.py#L1136-L1155

As you can see in this function, we should allow users to update all nested params if they're listed in `FIELDS_ALLOWED_TO_UPDATE` and `FIELDS_WITH_NESTED_FIELDS `

Reply:
Same goes for this check

https://github.com/valory-xyz/open-autonomy/blob/main/autonomy/configurations/base.py#L369-L374

Reply:
Yeah agree - NESTED_FIELDS_ALLOWED_TO_UPDATE is not helpful

Reply:
For `AgentConfig` we also need to update `FIELDS_WITH_NESTED_FIELDS`, to include `logging_config` I think